src/store/writer.rs

```diff
@@ -56,17 +91,26 @@ impl StoreWriter {
         self.intermediary_buffer.capacity() + self.current_block.capacity()
     }
 
-    /// Store bytes of a serialized document.
-    ///
-    /// The document id is implicitely the current number
-    /// of documents.
-    pub fn store_bytes(&mut self, serialized_document: &[u8]) -> io::Result<()> {
-        let doc_num_bytes = serialized_document.len();
-        VInt(doc_num_bytes as u64).serialize(&mut self.current_block)?;
-        self.current_block.write_all(serialized_document)?;
-        self.doc += 1;
+    /// Checks if the current block is full, and if so, compresses and flushes it.
+    fn check_flush_block(&mut self) -> io::Result<()> {
         if self.current_block.len() > self.block_size {
-            self.write_and_compress_block()?;
+            self.send_current_block_to_compressor()?;
+        }
+        Ok(())
+    }
+
+    /// Flushes current uncompressed block and sends to compressor.
+    fn send_current_block_to_compressor(&mut self) -> io::Result<()> {
```

Comment:
I think we can return the SendError directly here.
See discussion on call site.